File: jsimple-oauth/src/main/java/jsimple/oauth/builder/api/LiveOauthApi.java
package jsimple.oauth.builder.api;

import jsimple.oauth.extractors.AccessTokenExtractor;
import jsimple.oauth.extractors.JsonTokenExtractor;
import jsimple.oauth.model.OAuthConfig;
import jsimple.oauth.utils.OAuthEncoder;

public class LiveOAuthApi extends DefaultOAuthApi20 {
    @Override public String getAccessTokenEndpoint() {
        return "https://oauth.live.com/token?grant_type=authorization_code";
    }

    @Override public String getAuthorizationUrl(OAuthConfig config) {
        // Old code:
        // Preconditions.checkValidUrl(config.getCallback(), "Must provide a valid url as callback. Live does not support OOB");
        assert config.getCallback() != null && !config.getCallback().isEmpty() :
                "Must provide a valid url as callback. Live does not support OOB";

        String url = "https://oauth.live.com/authorize?client_id=" + config.getApiKey() + "&redirect_uri=" +
                OAuthEncoder.encode(config.getCallback()) + "&response_type=code";
        // Append scope if present
        String scope = config.getScope();
        if (scope != null)
            url += OAuthEncoder.encode(scope);

        return url;
    }

    @Override public AccessTokenExtractor getAccessTokenExtractor() {
        return new JsonTokenExtractor();
    }
}

File: jsimple-oauth/c#/jsimple/oauth/builder/api/LiveOAuthApi.cs
using System.Diagnostics;

namespace jsimple.oauth.builder.api
{

	using AccessTokenExtractor = jsimple.oauth.extractors.AccessTokenExtractor;
	using JsonTokenExtractor = jsimple.oauth.extractors.JsonTokenExtractor;
	using OAuthConfig = jsimple.oauth.model.OAuthConfig;
	using OAuthEncoder = jsimple.oauth.utils.OAuthEncoder;

	public class LiveOAuthApi : DefaultOAuthApi20
	{
		public override string AccessTokenEndpoint
		{
			get
			{
				return "https://oauth.live.com/token?grant_type=authorization_code";
			}
		}

		public override string getAuthorizationUrl(OAuthConfig config)
		{
			// Old code:
			// Preconditions.checkValidUrl(config.getCallback(), "Must provide a valid url as callback. Live does not support OOB");
			Debug.Assert(config.Callback != null && config.Callback.Length > 0, "Must provide a valid url as callback. Live does not support OOB");

			string url = "https://oauth.live.com/authorize?client_id=" + config.ApiKey + "&redirect_uri=" + OAuthEncoder.encode(config.Callback) + "&response_type=code";
			// Append scope if present
			string scope = config.Scope;
			if (scope != null)
				url += OAuthEncoder.encode(scope);

			return url;
		}

		public override AccessTokenExtractor AccessTokenExtractor
		{
			get
			{
				return new JsonTokenExtractor();
			}
		}
	}
}